File: sample-legacy/src/main/java/com/example/legacy/UserService.java
package com.example.legacy;

import java.util.ArrayList;
import java.util.List;

public class UserService {
    private final List<User> userList = new ArrayList<>();

    public void addUser(User user) {
        if (user == null) {
            throw new IllegalArgumentException("User cannot be null");
        }
        if (user.getId() == null || user.getId().isEmpty()) {
            throw new IllegalArgumentException("User ID cannot be null or empty");
        }
        userList.add(user);
    }

    public User findById(String id) {
        if (id == null) {
            return null;
        }
        for (User user : userList) {
            if (id.equals(user.getId())) {
                return user;
            }
        }
        return null;
    }

    public List<User> filterAdults() {
        List<User> result = new ArrayList<>();
        for (User user : userList) {
            if (user != null && user.getAge() >= 18) {
                result.add(user);
            }
        }
        return result;
    }

    public String formatUserSummary(User user) {
        if (user == null) {
            return "N/A";
        }
        StringBuilder sb = new StringBuilder();
        sb.append(user.getUsername() != null ? user.getUsername() : "Anonymous");
        sb.append(" (");
        sb.append(user.getEmail() != null ? user.getEmail() : "no-email");
        sb.append(")");
        return sb.toString();
    }
}

File: archived-runs/run-2-simpletodo-task-success/migrated-src/com/example/legacy/UserService.kt
package com.example.legacy

class UserService {
    private val userList = mutableListOf<User>()

    fun addUser(user: User?) {
        if (user == null) {
            throw IllegalArgumentException("User cannot be null")
        }
        if (user.id.isNullOrEmpty()) {
            throw IllegalArgumentException("User ID cannot be null or empty")
        }
        userList.add(user)
    }

    fun findById(id: String?): User? {
        if (id == null) return null
        return userList.firstOrNull { it.id == id }
    }

    fun filterAdults(): List<User> {
        return userList.filter { it.age >= 18 }
    }

    fun formatUserSummary(user: User?): String {
        if (user == null) return "N/A"
        val username = if (!user.username.isNullOrEmpty()) user.username else "Anonymous"
        val email = if (!user.email.isNullOrEmpty()) user.email else "no-email"
        return "$username ($email)"
    }
}